docs/conf.py

```diff
@@ -23,13 +23,27 @@
 templates_path = ["_templates"]
 source_suffix = ".rst"
 master_doc = "index"
-language = None
+language = "en"
 exclude_patterns = ["_build", "Thumbs.db", ".DS_Store"]
 pygments_style = None
 
 # Autodocumentation of modules
 extensions.append("sphinx.ext.autodoc")
 autodoc_typehints = "description"
+# For the life of me I can't get the docs to compile in nitpicky mode without these
+# ignore statements. You are most welcome to try and remove them.
+nitpick_ignore = [
+    ("py:class", "Config"),
+    ("py:class", "Action"),
+    ("py:class", "ActionTemplate"),
+    ("py:class", "Filter"),
+    ("py:class", "t.List"),
+    ("py:class", "t.Tuple"),
+    ("py:class", "t.Any"),
+    ("py:class", "tutor.hooks.filters.P"),
+    ("py:class", "tutor.hooks.filters.T"),
+    ("py:class", "tutor.hooks.actions.P"),
+]
```

Comment:
@regisb I banged my head against this for a while:

* With the PR as-is, you can remove `Action`, `Filter`, and `t.Any` from the ignore list.
* In hooks/consts.py, if you write `from typing import List, Tuple` instead of `import typing as t` (and update references accordingly), then the warnings for `t.List` and `t.Tuple` go away.
  * Alternatively, the docs build fine if you use `from __future__ import annotations` and then replace the types with the built-ins `list` and `tuple`.
* I could not figure out any way to resolve the warnings for `ActionTemplate`, `T`, or either `P`.

My non-expert guess is that Sphinx is getting confused by one or both of these things:
* things that are *types* but aren't *classes* (such as `Config`, `List`, `P`, etc.)
* things that use more than one canonical import path, for example:
  * `typing.List` being renamed to `t.List`
  * `tutor.hooks.consts.Actions` being re-exported as `tutor.hooks.Actions`

Reply:
> Alternatively, the docs build fine if you use from __future__ import annotations and then replace the types with the built-ins list and tuple.

OK this is really cool. I'm adding the built-ins now. We are only left with a couple nitpick-ignored classes now.